### PR TITLE
[OCPSTRAT-506] Azure Managed Identity (Workload Identity) Support

### PR DESCRIPTION
**Pull request by huangmingxia**
Azure Managed Identity (Workload Identity) Support
[CCO-380](https://issues.redhat.com/browse/CCO-380) 
[CCO-407](https://issues.redhat.com/browse/CCO-407)

### CODE REVIEW

**Comment by huangmingxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-workload-identity-p1-f3

**Comment by huangmingxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-workload-identity-p1-f3

**Comment by huangmingxia** [comment]
@yunjiang29 @jianlinliu Could you help review this pr when you free? Thanks for your time.

File: ci-operator/step-registry/ipi/conf/featureset/ipi-conf-featureset-commands.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/featureset/ipi-conf-featureset-commands.sh:1]: We do not need this step to set `TechPreviewNoUpgrade`, but set `FEATURE_SET: TechPreviewNoUpgrade` env in job config level.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 2]
```diff
@@ -1626,6 +1626,17 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-azure-ipi-ovn-ipsec-azurefile-csi
+- as: azure-ipi-ovn-workload-identity-p1-f3
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:1629]: `f3` ? I think it is to aggressive, this feature sounds like a TP feature, I think `f14`  is enough.

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:1629]: And if it is TP feature, pls add `tp` keyword in the job name, also pls remove `ovn` keyword from its name.

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:1629]: Also need to add the similar job in arm nightly ci.

Comment (huangmingxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:1629]: @jianlinliu  Thank you very much, I agree with your opinion, but at the beginning we need to debug to make this ci reach a relatively stable state. After stabilization, the frequency will be adjusted to f14 or f28. 
If f3 is to aggressive, what about f7?

File: ci-operator/step-registry/ipi/conf/featureset/ipi-conf-featureset-commands.sh (hunk 1 above)
Comment (huangmingxia) [reply ci-operator/step-registry/ipi/conf/featureset/ipi-conf-featureset-commands.sh:1]: @jianlinliu Thanks, didn't notice this variable.

**Comment by huangmingxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-workload-identity-tp-p1-f7

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml (hunk 2 above)
Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml]: Now we can trigger any job via remote api, https://github.com/openshift/release/blob/master/ci-operator/config/openshift/openshift-tests-private/README.md#trigger-a-new-test-for-an-existing-periodic-job-in-prow, so I do not think it is a problem for your debugging. You can trigger it any time on your demand.

And from the whole team management review, we add many jobs every day, if each team think it is important to have a ci resulta in short days, then our quota would be easily run out, unless there is a strong reason to set an aggressive test frequency. 

If we set f14 for both arm and amd, then the job would be executed 1 time in 1 week on average, I think it is reasonable enough for TP feature job.

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/provision/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-provision-chain.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+chain:
+  as:  cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-provision
+  steps: 
+  - chain: ipi-conf-azure
+  - ref: ipi-conf-manual-creds
+  - ref: ipi-conf-azure-oidc-creds-provision
+  - ref: ipi-conf-azure-provisioned-resourcegroup
```
[the remark this answers is not in the document]
Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/provision/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-provision-chain.yaml:7]: or @huangmingxia can you confirm if `ccoctl azure create-all` command in `ipi-conf-azure-oidc-creds-provision` would create the `"${CLUSTER_NAME}"` resource group on azure?

[the remark this answers is not in the document]
Comment (huangmingxia) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/provision/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-provision-chain.yaml:7]: @jianlinliu Yes，`ccoctl azure create-all` can create resource group on azure.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml (hunk 2 above)
Comment (huangmingxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml]: @jianlinliu Thanks for your information!! If so, agree to tune to f14.
 I can use this https://github.com/openshift/release/blob/master/ci-operator/config/openshift/openshift-tests-private/README.md#trigger-a-new-test-for-an-existing-periodic-job-in-prow

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/provision/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-provision-chain.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (huangmingxia) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/provision/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-provision-chain.yaml:7]: @jianlinliu 
The `azure-oidc-creds-provision-command.sh` is using `ccoctl azure create-all` to create the  resource group, `${CLUSTER_NAME}` is also the `azure_resourcegroup` name.

This is a small snippet of logs
`2023/07/12 08:13:59 Created federated identity credential /subscriptions/xxxx/resourcegroups/ci-op-igvi1gwb-6e310-oidc/` the resource group name is `ci-op-igvi1gwb-6e310`.

Login to the azure client, can see this rescource group is create successfully on azure.
 
image
image


Login to this cluster:
$ oc whoami --show-console
https://console-openshift-console.apps.ci-op-igvi1gwb-6e310.qe.azure.devcluster.openshift.com

[the remark this answers is not in the document]
Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/provision/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-provision-chain.yaml:7]: hmm, who created `ci-op-igvi1gwb-6e310-oidc`?

[the remark this answers is not in the document]
Comment (huangmingxia) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/provision/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-provision-chain.yaml:7]: @jianlinliu Yes, `ci-op-igvi1gwb-6e310-oidc` is also created by `ccoctl azure create-all`. 
`ccoctl azure create-all` is used to create OIDC issuer and managed identities.

[the remark this answers is not in the document]
Comment (huangmingxia) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/provision/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-provision-chain.yaml:7]: `ccoctl azure create-all` combines `create-key-pair`, `create-oidc-issuer` and `create-managed-identities` into a single command.

**Comment by MayXuQQ** [comment]
>  if we do not create the resource group in advance, but directly set it in install-config.yaml, installer can help create it?
No. 
FATAL failed to fetch Terraform Variables: failed to fetch dependency of "Terraform Variables": failed to generate asset "Platform Provisioning Check": platform.azure.resourceGroupName: Internal error: failed to get resource group: failed to get resource group: resources.GroupsClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceGroupNotFound" Message="Resource group 'maxu-rg' could not be found."

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/provision/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-provision-chain.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/provision/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-provision-chain.yaml:7]: got it.

**Comment by huangmingxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-azure-ipi-workload-identity-tp-p1-f14

**Comment by huangmingxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-workload-identity-tp-p1-f14

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml (hunk 2 above)
Comment (huangmingxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml]: Hi @jianlinliu, arm nightly ci add down. Could you help review again? Thanks.

**Comment by jianlinliu** [comment]
cc @jinyunma @liangxia to review

**Comment by jianlinliu** [comment]
@huangmingxia the robot said the PR need rebase.

**Comment by liangxia** [comment]
One of the required job failed, `ci/prow/generated-config — Job failed.`

Looks like you forgot to run `make update`

File: ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14-periodics.yaml [hunk 4]
```diff
@@ -27600,16 +27600,6 @@ periodics:
     job-release: "4.14"
     pj-rehearse.openshift.io/can-be-rehearsed: "true"
   name: periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-disconnected-private-p3-f28-long-duration-cloud
-  reporter_config:
```
Comment (liangxia) [inline ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14-periodics.yaml:27603]: Unexpected removal of existing configs.

Comment (huangmingxia) [reply ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14-periodics.yaml]: Sorry, I thought it was removed by someone else, I don't have sync here.

**Pull request merged**